src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Thread.NativeAot.cs

```diff
@@ -483,6 +483,41 @@ private static void StopThread(Thread thread)
             }
         }
 
+
+        private static int StackSizeFromConfig
+        {
+            get => _stackSizeFromConfig ??= GetDefaultStackSize();
```

Comment:
This pattern with `Nullable<int>` is not thread safe. Nullable<int> is a struct with two fields. This pattern has a race condition caused by tearing.

I do not think you need to cache the value on the managed side. The value is cached on the unmanaged side. You can call every time.